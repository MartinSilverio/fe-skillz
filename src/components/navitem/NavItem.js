import React, { useState } from 'react';
import NavList from '../navlist/NavList';
import { NavLink, useHistory } from 'react-router-dom';

import './NavItem.scss';

function NavItem({ navItem }) {
  const [focused, setFocused] = useState(false);
  const history = useHistory();

  const handleFocus = () => {
    setFocused(true);
  };
  const handleBlur = () => {
    setFocused(false);
  };

  const handleKeyPress = (e) => {
    //Handle space; enter works by default
    if (e.key === ' ') {
      history.push(navItem.url);
    }
  };

  return (
    <li role='none' className='nav-item'>
      <NavLink
        to={navItem.url}
        role='menuitem'
        aria-haspopup={navItem.children ? true : false}
        aria-expanded={navItem.children && focused ? true : false}
        activeClassName='selected'
        isActive={(match, location) => location.pathname.includes(navItem.url)}
        onFocus={handleFocus}
        onBlur={handleBlur}
        onKeyPress={handleKeyPress}
      >
        {navItem.title}
      </NavLink>
      {navItem.children && (
        <NavList
          className='subnav'
          role='menu'
          aria-label={navItem.title}
          navItems={navItem.children}
        />
      )}
    </li>
  );
}

export default NavItem;
