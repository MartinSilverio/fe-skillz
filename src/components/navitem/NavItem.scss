@import '../../colors';

.subnav {
  position: absolute;
  left: 30px;
  top: 50px;
  padding: 0;
  list-style: none;
  height: 1px;
  width: 1px;
  overflow: hidden;
  clip: rect(1px 1px 1px 1px); /* IE6, IE7 */
  clip: rect(1px, 1px, 1px, 1px);
}

.nav-item {
  position: relative;
  height: 100%;
  display: grid;
  place-items: center;
  padding: 0 50px;

  &:hover {
    color: $gold;
    cursor: pointer;
  }
  &:hover .subnav {
    color: $white;
    padding: 0.5rem;
    width: 9rem;
    height: auto;
    background: $mediumGrey;
    clip: auto;
  }
}
