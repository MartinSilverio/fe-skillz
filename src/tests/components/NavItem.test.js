import React from 'react';
import { shallow } from 'enzyme';
import NavItem from '../../components/navitem/NavItem';

const itemWithNoChild = {
  title: 'Amet',
  id: 2,
  selected: false,
  url: '/amet',
};

const itemWithChild = {
  title: 'Consectetur',
  id: 3,
  selected: false,
  url: '/consectetur',
  children: [
    {
      title: 'Tempor',
      id: 33,
      selected: false,
      url: '/consectetur/tempor',
    },
  ],
};

describe('NavItem', () => {
  it('Should utilize NavList if navItem has children', () => {
    expect.assertions(1);
    expect(shallow(<NavItem navItem={itemWithChild} />)).toMatchSnapshot();
  });
  it('Should not have NavList if navItem has no children', () => {
    expect.assertions(1);
    expect(shallow(<NavItem navItem={itemWithNoChild} />)).toMatchSnapshot();
  });
});
