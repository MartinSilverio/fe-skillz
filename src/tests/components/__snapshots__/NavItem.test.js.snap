// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavItem Should not have NavList if navItem has no children 1`] = `
<li
  className="nav-item"
  role="none"
>
  <NavLink
    activeClassName="selected"
    aria-haspopup={false}
    isActive={[Function]}
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyPress={[Function]}
    role="menuitem"
    to="/amet"
  >
    Amet
  </NavLink>
</li>
`;

exports[`NavItem Should utilize NavList if navItem has children 1`] = `
<li
  className="nav-item"
  role="none"
>
  <NavLink
    activeClassName="selected"
    aria-expanded={false}
    aria-haspopup={true}
    isActive={[Function]}
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyPress={[Function]}
    role="menuitem"
    to="/consectetur"
  >
    Consectetur
  </NavLink>
  <NavList
    aria-label="Consectetur"
    className="subnav"
    navItems={
      Array [
        Object {
          "id": 33,
          "selected": false,
          "title": "Tempor",
          "url": "/consectetur/tempor",
        },
      ]
    }
    role="menu"
  />
</li>
`;
